src/java/us/kbase/mobu/ModuleBuilder.java

```diff
@@ -360,24 +361,31 @@ private static int runTestCommand(TestCommandArgs testArgs, JCommander jc) {
     }
 
     private static void printVersion() {
-        String gitCommit = getGitCommit();
-        System.out.println("KBase SDK version " + VERSION + (gitCommit == null ? "" : (" (commit " + gitCommit + ")")));
+        String gitCommit = getGitProp("commit");
+        String timestamp = getGitProp("build_timestamp");
+        System.out.println("KBase SDK version " + VERSION
+            + (timestamp == null ? "" : ("_" + timestamp))
+            + (gitCommit == null ? "" : (" (commit " + gitCommit + ")")));
     }
 
-    public static String getGitCommit() {
-        String gitCommit = null;
+
+    public static String getGitProp(String propertyToGet) {
+        String propertyValue = null;
         try {
             Properties gitProps = new Properties();
             InputStream is = ModuleBuilder.class.getResourceAsStream("git.properties");
             gitProps.load(is);
             is.close();
-            gitCommit = gitProps.getProperty("commit");
-        } catch (Exception ignore) {}
-        return gitCommit;
+            propertyValue = gitProps.getProperty(propertyToGet);
+        } catch (Exception e) {
+            showError("Error while retrieving version information", e.getMessage());
```

Comment:
`showError` just prints the error, it doesn't exit. I think you'll have to do something else here to stop execution.

Reply:
I don't think it's necessary to stop execution if the `git.properties` file can't be found or these properties aren't available -- it's basically just adding in some extra "nice to have" information, it isn't mission critical. The properties file is generated during the build process and deleting it afterwards had no effect on the functioning of kb-sdk.

Reply:
But why not? If it's not there, clearly something went wrong with the build. Reduce runtime variables IMO.

Reply:
My thinking is more along the lines of "I don't want to give kb-sdk a reason to die unexpectedly", plus I couldn't create a situation manually where kb-sdk built successfully but the `git.properties` file was not present.

Reply:
I'm thinking I don't want unexpected behavior from the SDK when something's broken. I want it to fail fast.

> plus I couldn't create a situation manually where kb-sdk built successfully but the git.properties file was not present.

Then failing won't cause any problems in practice unless something is really wrong, so might as well fail.

Reply:
OK, added a `System.exit(1)` after emitting the error message.